### PR TITLE
Adding log to report containers registry status for OCP workloads

### PR DESCRIPTION
**Pull request by vishnuchalla**
## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] Optimization
- [ ] Documentation Update

## Related Tickets & Documents

- Related Issue # https://github.com/cloud-bulldozer/kube-burner/issues/389
- Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
Tested on OCP cluster using the ocp wrapper command

```
[vchalla@vchalla kube-burner]$ kube-burner ocp node-density-heavy --timeout 1h --metrics-endpoint ~/metrics-endpoints.yaml --pods-per-node=50 --log-level=debug 2>&1 | tee -a kube-burner-$(date +"%F_%H-%M-%S")
time="2023-08-06 00:49:16" level=info msg="Cluster image registry is up and running and ready to use with a healthy state" file="utils.go:139"
time="2023-08-06 00:49:17" level=info msg="File node-density-heavy.yml available in the current directory, using it" file="helpers.go:178"
time="2023-08-06 00:49:17" level=info msg="👽 Initializing prometheus client with URL: https://prometheus-k8s-openshift-monitoring.apps.vchalla-clair-test-8.perfscale.devcluster.openshift.com" file="prometheus.go:43"
time="2023-08-06 00:49:18" level=info msg="🔥 Starting kube-burner (imageregistry@c193de1b8083a54d1ec1e1893af65ed55c4be275) with UUID 7d6b4099-9e66-4392-a19f-341c27613fc6" file="job.go:83"
time="2023-08-06 00:49:18" level=info msg="📈 Creating measurement factory" file="factory.go:51"
```

### CODE REVIEW

File: pkg/burner/utils.go [hunk 1]
```diff
@@ -102,6 +104,44 @@ func (ex *Executor) Verify() bool {
...
+		if condition.Type == "Progressing" && condition.Status == "False" && condition.Reason == "Ready" {
+			logMessage += " and ready to use"
+		}
+		if condition.Type == "Degraded" && condition.Status == "False" && condition.Reason == "AsExpected" {
+			logMessage += " with a healthy state"
+		}
+	}
+	if readyFlag {
+		log.Infof("Cluster image registry is %s", logMessage)
+	} else {
```
Comment (smalleni) [inline pkg/burner/utils.go:140]: Workloads like cluster-density depend on some image registry being configured, what is your plan after checking for registry?

Comment (rsevilla87) [reply pkg/burner/utils.go:140]: IMO, we should call this function only from workloads having Build or BuildConfigs. And hang if the registry is not ready

Comment (vishnuchalla) [reply pkg/burner/utils.go:140]: Yes. Changed as suggested!

File: pkg/workloads/cluster-density.go [hunk 2]
```diff
@@ -31,6 +32,9 @@ func NewClusterDensity(wh *WorkloadHelper, variant string) *cobra.Command {
 		Use:   variant,
 		Short: fmt.Sprintf("Runs %v workload", variant),
 		PreRun: func(cmd *cobra.Command, args []string) {
+			if !burner.VerifyContainerRegistry(wh.restConfig) {
+				os.Exit(0)
```
Comment (rsevilla87) [inline pkg/workloads/cluster-density.go:36]:
```suggestion
				os.Exit(1)
```

Comment (vishnuchalla) [reply pkg/workloads/cluster-density.go]: Changed as suggested.

**Review by rsevilla87 — APPROVED** [review]
lgtm

**Pull request merged**